### PR TITLE
Add two factor auth support for Radar

### PR DESCRIPTION
**Pull request by keith**

### CODE REVIEW

File: Sources/Sonar/APIs/AppleRadar.swift [hunk 1]
```diff
@@ -24,37 +24,39 @@ final class AppleRadar: BugTracker {
     /**
      Login into radar by an apple ID and password.
 
-     - parameter closure: A closure that will be called when the login is completed, on success it will
-                          contain a list of `Product`s; on failure a `SonarError`.
+     - parameter getTwoFactorCode: A closure to retrieve a two factor auth code from the user
```
Comment (sberrevoets) [inline Sources/Sonar/APIs/AppleRadar.swift]: Nit: missing period :trollface:

File: Sources/Sonar/APIs/BugTracker.swift [hunk 2]
```diff
@@ -18,10 +18,12 @@ protocol BugTracker {
     /**
      Login into bug tracker. This method will use the authentication information provided by the service enum.
 
-     - parameter closure: A closure that will be called when the login is completed,
-                          on failure a `SonarError`.
+     - parameter getTwoFactorCode: A closure to retrieve a two factor auth code from the user
```
Comment (sberrevoets) [inline Sources/Sonar/APIs/BugTracker.swift]: Same

File: Sources/Sonar/Sonar.swift [hunk 3]
```diff
@@ -44,12 +47,16 @@ public class Sonar {
     /**
      Similar to `create` but logs the user in first.
 
-     - parameter radar:   The radar model with the information for the ticket.
-     - parameter closure: A closure that will be called when the login is completed, on success it will
-                          contain a radar ID; on failure a `SonarError`.
+     - parameter radar:            The radar model with the information for the ticket.
+     - parameter getTwoFactorCode: A closure to retrieve a two factor auth code from the user
```
Comment (sberrevoets) [inline Sources/Sonar/Sonar.swift]: Same

File: Sources/Sonar/Sonar.swift [hunk 4]
```diff
@@ -17,11 +17,14 @@ public class Sonar {
     /**
      Login into bug tracker. This method will use the authentication information provided by the service enum.
 
-     - parameter closure: A closure that will be called when the login is completed,
-                          on failure a `SonarError`.
+     - parameter getTwoFactorCode: A closure to retrieve a two factor auth code from the user
```
Comment (sberrevoets) [inline Sources/Sonar/Sonar.swift]: Same

File: Sources/Sonar/APIs/AppleRadarRouter.swift [hunk 5]
```diff
@@ -17,6 +17,8 @@ Apple's radar request router.
 enum AppleRadarRouter {
     case Products(CSRF: String)
     case Login(appleID: String, password: String)
+    case AuthorizeTwoFactor(code: String, scnt: String, sessionID: String)
+    case FetchCSRF
     case Create(radar: Radar, CSRF: String)
     case ViewProblem
```
Comment (Reflejo) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: lower case all these 💃

Comment (keith) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: next PR!

File: Sources/Sonar/Sonar.swift [hunk 6]
```diff
@@ -44,12 +47,16 @@ public class Sonar {
...
-     - parameter closure: A closure that will be called when the login is completed, on success it will
-                          contain a radar ID; on failure a `SonarError`.
+     - parameter radar:            The radar model with the information for the ticket.
+     - parameter getTwoFactorCode: A closure to retrieve a two factor auth code from the user
+     - parameter closure:          A closure that will be called when the login is completed, on success it
+                                   will contain a radar ID; on failure a `SonarError`.
     */
-    public func loginThenCreate(radar: Radar, closure: @escaping (Result<Int, SonarError>) -> Void) {
-        self.tracker.login { result in
+    public func loginThenCreate(
```
Comment (Reflejo) [inline Sources/Sonar/Sonar.swift]: can you update the example if we are changing this signature?

Comment (keith) [inline Sources/Sonar/Sonar.swift]: yes

**Comment by keith** [comment]
Pushed a second commit that has the README and example updates for the new APIs. They've changed a lot since the last time we've compiled that.

**Pull request merged**